extension-ui/background.js

```diff
@@ -193,3 +193,198 @@ function showNotification(site, num, options = {}) {
   
   chrome.notifications.create(notificationOptions);
 }
+
+let timeoutId = null;
+let notificationTimeoutId = null;
```

Comment:
you have to manage the data in dictionary according to site url, because multiple limitations can be at the same time
add class with all the properties and create instance for each limited site, save it in dictionary according to site url